### PR TITLE
Fix git script

### PR DESCRIPTION
**Pull request by go-kipi**

### CODE REVIEW

File: installOrchestDDocker.sh [hunk 1]
```diff
@@ -55,37 +91,35 @@ under "new ssh key"'
...
     read -p "> " INPSEL
 	case $INPSEL in
 	    "1")
-            show "https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#setting-your-email-address-for-every-repository-on-your-computer"
-            exit
+            printf "{\n\t\"server\":\"$GITPATH\",\n\t\"gitEmail\":\"$gitemail\",\n\t\"devBranch\": \"main\",\n\t\"lockedBranches\":[\"dev\",\"master\",\"main\"]\n}\n" > ${userPath}/settings/git.json
 			;;
 	    "2")
-            exit
+	    readGithubUser;
```
Comment (guybrand) [inline installOrchestDDocker.sh]: email first
+github url
"please type the github url you will be working on for this development project:
(if its youre private github, its probably https://github/{user}, if its a company shared github, its probably something like https://github/mycompany)
 ">https://github.com/"
this means setting would have
server
email
user

File: installOrchestDDocker.sh [hunk 2]
```diff
@@ -28,14 +29,49 @@ function show {
 	echo -e "${fnt}${txt}`tput sgr0`"
 }
 
-function installSshKey() {
+function readGithubUser {
```
Comment (guybrand) [inline installOrchestDDocker.sh]: requestGithubUser+GithubEmail

File: installOrchestDDocker.sh [hunk 3]
```diff
@@ -28,14 +29,49 @@ function show {
...
+      show "please type your github user (not email), i.e. if your github path is
+https://github.com/leonardo-da-vinci
+please type:
+leonardo-da-vinci"
+          read -p '> ' gituser
+}
+
+function readGithubEmail {
+    show "please type your github email"
```
Comment (guybrand) [inline installOrchestDDocker.sh]: e.g. mymail@gmail.com / mymail@mycompany.com

File: installOrchestDDocker.sh [hunk 4]
```diff
@@ -28,14 +29,49 @@ function show {
 	echo -e "${fnt}${txt}`tput sgr0`"
 }
 
-function installSshKey() {
+function readGithubUser {
+      show "please type your github user (not email), i.e. if your github path is
```
Comment (guybrand) [inline installOrchestDDocker.sh]: please type your github user, (github -> top right corner, "Signed in as xxx")

File: installOrchestDDocker.sh [hunk 5]
```diff
@@ -55,37 +91,35 @@ under "new ssh key"'
 }
 
 #  make sure you have an active git user on either GH or BBKT
-show "checking git config user.email:"
-git config user.email
-if [ $? -ne 0 ]; then
-show "Need to config git  user.email"
-show "Show you how?\n[1]Yes\n[2]No"
+show "checking git config user.email..."
+gitemail=$(git config user.email)
```
Comment (guybrand) [inline installOrchestDDocker.sh]: move gitemail=$(git config user.email)

to readGithubEmailFromEnv

File: installOrchestDDocker.sh [hunk 6]
```diff
@@ -28,14 +29,49 @@ function show {
...
+          read -p '> ' gituser
+}
+
+function readGithubEmail {
+    show "please type your github email"
+        read -p "> " gitemail
+}
+
+
+function readGithubMailIfNotExist {
```
Comment (guybrand) [inline installOrchestDDocker.sh]: readGithubEmailFromEnv

File: installOrchestDDocker.sh [hunk 7]
```diff
@@ -28,14 +29,49 @@ function show {
...
+    if [ $? -ne 0 ]; then
+    show "Need to config git  user.email"
```
Comment (guybrand) [inline installOrchestDDocker.sh]: how do I find it ?  https://github.com/settings/emails

**Review by guybrand — CHANGES_REQUESTED** [review]

File: installOrchestDDocker.sh [hunk 8]
```diff
@@ -7,6 +7,7 @@ cd ..
 reset;
 
 origpath=$PWD
```
Comment (guybrand) [inline installOrchestDDocker.sh]: line #1

File: installOrchestDDocker.sh [hunk 9]
```diff
@@ -28,64 +29,113 @@ function show {
...
-function installSshKey() {
-    show " it looks like you dont have git cli ssh key, which is the secured way to connect to github how would you like to proceed ?"
-    show "[1]show me the bash script and I will run it  \n[2]I will do it manually"
-        read -p "> " INPSEL
-    	case $INPSEL in
-    	    "1")
-show 'ssh-keygen -t ed25519 -C "xxxx@gmail.com"
-Enter -> Enter (empty passphrase)
+function requestGithubUser {
+      show 'please type your github user, (github -> top right corner, "Signed in as xxx"):'
```
Comment (guybrand) [inline installOrchestDDocker.sh]: goto github.com

File: installOrchestDDocker.sh [hunk 10]
```diff
@@ -28,64 +29,113 @@ function show {
...
+function requestGithubEmail {
+    show "please type your github email, e.g. mymail@gmail.com / mymail@mycompany.com :"
+        read -p "> " gitemail
+}
+
+function readGithubEmailFromEnv {
+  show "checking git config user.email..."
+  gitemail=$(git config user.email)
+  if [ $? -ne 0 ]; then
+  show "how do I find it ? https://github.com/settings/emails"
```
Comment (guybrand) [inline installOrchestDDocker.sh]: move to request

**Review by guybrand — CHANGES_REQUESTED** [review]
fill in all 3 git parameter to memory
if not ALLLL exist - ask to fill in alll
if all exist - prompt if these are the parameter -> 
  yes - use the,
  no - fill in as if read git conf retuirned empty

File: installOrchestDDocker.sh [hunk 11]
```diff
@@ -28,64 +30,112 @@ function show {
...
-	    "1")
-            show "https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#setting-your-email-address-for-every-repository-on-your-computer"
-            exit
-			;;
-	    "2")
-            exit
-			;;
-	esac
-fi
+    checkGitCliSshKey
```
Comment (guybrand) [inline installOrchestDDocker.sh]: move after clone apispecs + 
cd apispecs
git config user.name=
git config user.email=

File: installOrchestDDocker.sh [hunk 12]
```diff
@@ -28,64 +30,112 @@ function show {
...
-            fi
-
-GITPATH="https://github.com/$GITUSER"
-
-userPath=/home/$USER/orchestD
+if [[ "$gitemail" == "" || "$gituser" == "" ]];
+then
+  	requestGithubEmail
+  	requestGithubUser
+  	requestGithubUrl
```
Comment (guybrand) [inline installOrchestDDocker.sh]: move down and remove from confirmeGitParams

**Review by guybrand — CHANGES_REQUESTED** [review]
fill in all 3 git parameter to memory
if not ALLLL exist - ask to fill in alll
if all exist - prompt if these are the parameter -> 
  yes - use the,
  no - fill in as if read git conf retuirned empty

File: integrations/install.sh [hunk 13]
```diff
@@ -17,68 +17,68 @@ function show {
 	echo -e "${fnt}${txt}`tput sgr0`"
 }
 
+# checking Prerequisites
+function goPrerequisites {
+    which go
+    if [ $? -ne 0 ]; then
+    show "Need to install Go"
```
Comment (guybrand) [inline integrations/install.sh]: please install golang + link to dnd

File: integrations/install.sh [hunk 14]
```diff
@@ -17,68 +17,68 @@ function show {
...
-	    "2")
-            exit
-			;;
-	esac
-fi
-show "done"
+    goV=`go version | { read _ _ v _; echo ${v#go}; }`
+    echo $goV
+    if [[ "$goV" < "1.19.0" ]]; then
+      echo "supported on golang 1.19+"
```
Comment (guybrand) [inline integrations/install.sh]: minimal supported version og golang is 1.19 (+link)

File: integrations/install.sh [hunk 15]
```diff
@@ -17,68 +17,68 @@ function show {
...
+      exit
+    fi
+}
+
+function gitPrerequisites {
+        which git
+        if [ $? -ne 0 ]; then
+        show "Need to install Git"
```
Comment (guybrand) [inline integrations/install.sh]: Please install git + link

File: integrations/install.sh [hunk 16]
```diff
@@ -17,68 +17,68 @@ function show {
...
-	    "2")
-            exit
-			;;
-	esac
-fi
-show "done"
+function dockerPrerequisites {
+      which docker
+      if [ $? -ne 0 ]; then
+      show "Need to install Docker"
```
Comment (guybrand) [inline integrations/install.sh]: ...

File: integrations/install.sh [hunk 17]
```diff
@@ -17,68 +17,68 @@ function show {
...
+      exit
+      fi
+
+      docker -v
+}
 
+function dockerComposePrerequisites {
+      which docker-compose
+      if [ $? -ne 0 ]; then
+      show "Need to install docker-compose"
```
Comment (guybrand) [inline integrations/install.sh]: ...

File: integrations/install.sh [hunk 18]
```diff
@@ -17,68 +17,68 @@ function show {
...
+      exit
+      fi
+
+      docker-compose version
+}
+
+function mongoDBPrerequisites {
+      which mongo
+      if [ $? -ne 0 ]; then
+      show "Need to install mongo shell"
```
Comment (guybrand) [inline integrations/install.sh]: not a must, so
"for best experience, we advise you also install Mongo db tools" (+link)

File: integrations/install.sh [hunk 19]
```diff
@@ -17,68 +17,68 @@ function show {
...
+      exit
+      fi
+      mongo -version
+      if [ $? -ne 0 ]; then
+      show "Need to install mongo shell"
```
Comment (guybrand) [inline integrations/install.sh]: ?

**Review by guybrand — CHANGES_REQUESTED** [review]

**Review by orchestd — APPROVED** [review]

**Pull request merged**